rdkPlugins/Storage/source/MountOwnerDetails.cpp

```diff
@@ -0,0 +1,241 @@
+ */
+bool MountOwnerDetails::onCreateRuntime() const
+{
+    AI_LOG_FN_ENTRY();
+
+    bool success = false;
+    struct stat buffer;
+    if (stat(mMountOwnerProperties.source.c_str(), &buffer) == 0)
+    {
+        success = processOwners();
+    }
+    else
+    {
+        // Mount not found to ignore request to change ownership
+        success = true;
+        AI_LOG_INFO("Mount '%s' does not exist, change ownership skipped", mMountOwnerProperties.source.c_str());
+    }
+
+    AI_LOG_FN_EXIT();
+    return success;
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Get user and group IDs based on their configured
+ *
+ *  @param[out]  userId  ID corresponding to the configured user name
+ *  @param[out]  groupId  ID corresponding to the configured group name
+ *  @return true on success, false on failure
+ */
+bool MountOwnerDetails::getOwnerIds(uid_t& userId, gid_t& groupId) const
+{
+    struct passwd *pwd;
+    struct group *grp;
+    if (mMountOwnerProperties.user.empty())
+    {
+        AI_LOG_INFO("Using default user '%d'", mDefaultUserId);
+        userId = mDefaultUserId;
+    }
+    else
+    {
+        pwd = getpwnam(mMountOwnerProperties.user.c_str());
+        if (pwd == NULL)
+        {
+            AI_LOG_SYS_ERROR(errno, "User '%s' not found", mMountOwnerProperties.user.c_str());
+            return false;
+        }
+        else
+        {
+            userId = pwd->pw_uid;
+        }
+    }
+
+    if (mMountOwnerProperties.group.empty())
+    {
+        AI_LOG_INFO("Using default group '%d'", mDefaultGroupId);
+        groupId = mDefaultGroupId;
+    }
+    else
+    {
+        grp = getgrnam(mMountOwnerProperties.group.c_str());
+        if (grp == NULL)
+        {
+            AI_LOG_SYS_ERROR(errno, "Group '%s' not found", mMountOwnerProperties.group.c_str());
+            return false;
+        }
+        else
+        {
+            groupId = grp->gr_gid;
+        }
+    }
+
+    return true;
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Process individual mount owner and change ownership either singularly
+ * or recursively
+ *
+ *  @return true on success, false on failure
+ */
+bool MountOwnerDetails::processOwners() const
+{
+    AI_LOG_FN_ENTRY();
+
+    uid_t userId = 0;
+    gid_t groupId = 0;
+
+    if (getOwnerIds(userId, groupId))
+    {
+        if (mMountOwnerProperties.recursive)
+        {
+            if (!changeOwnerRecursive(mMountOwnerProperties.source, userId, groupId))
+            {
+                AI_LOG_ERROR("Failed to change owner recursively of '%s' to '%d:%d",
+                             mMountOwnerProperties.source.c_str(),
+                             userId,
+                             groupId);
+                return false;
+            }
+        }
+        else
+        {
+            changeOwner(mMountOwnerProperties.source, userId, groupId);
```

Comment:
Report the status of changeOwner (what if it failed?).

Reply:
Already reported in changeOwner itself.